includes/api/class-wc-rest-coupons-controller.php

```diff
@@ -0,0 +1,562 @@
+						'description' => __( 'Whether to bypass trash and force deletion.', 'woocommerce' ),
+					),
+				),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Query args.
+	 *
+	 * @param array $args
+	 * @param WP_REST_Request $request
+	 * @return array
+	 */
+	public function query_args( $args, $request ) {
+		global $wpdb;
+
+		if ( ! empty( $request['code'] ) ) {
+			$id = $wpdb->get_var( $wpdb->prepare( "SELECT id FROM $wpdb->posts WHERE post_title = %s AND post_type = 'shop_coupon' AND post_status = 'publish'", $request['code'] ) );
+
+			$args['post__in'] = array( $id );
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Prepare a single coupon output for response.
+	 *
+	 * @param WP_Post $post Post object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $data
+	 */
+	public function prepare_item_for_response( $post, $request ) {
+		global $wpdb;
+
+		// Get the coupon code.
+		$code = $wpdb->get_var( $wpdb->prepare( "SELECT post_title FROM $wpdb->posts WHERE id = %s AND post_type = 'shop_coupon' AND post_status = 'publish'", $post->ID ) );
+
+		$coupon = new WC_Coupon( $code );
+
+		$data = array(
+			'id'                           => $coupon->id,
+			'code'                         => $coupon->code,
+			'type'                         => $coupon->type,
+			'date_created'                 => wc_rest_prepare_date_response( $post->post_date_gmt ),
+			'date_modified'                => wc_rest_prepare_date_response( $post->post_modified_gmt ),
+			'amount'                       => wc_format_decimal( $coupon->coupon_amount, 2 ),
+			'individual_use'               => ( 'yes' === $coupon->individual_use ),
+			'product_ids'                  => array_map( 'absint', (array) $coupon->product_ids ),
+			'exclude_product_ids'          => array_map( 'absint', (array) $coupon->exclude_product_ids ),
+			'usage_limit'                  => ( ! empty( $coupon->usage_limit ) ) ? $coupon->usage_limit : null,
+			'usage_limit_per_user'         => ( ! empty( $coupon->usage_limit_per_user ) ) ? $coupon->usage_limit_per_user : null,
+			'limit_usage_to_x_items'       => (int) $coupon->limit_usage_to_x_items,
+			'usage_count'                  => (int) $coupon->usage_count,
+			'expiry_date'                  => ( ! empty( $coupon->expiry_date ) ) ? wc_rest_prepare_date_response( $coupon->expiry_date ) : null,
+			'enable_free_shipping'         => $coupon->enable_free_shipping(),
+			'product_category_ids'         => array_map( 'absint', (array) $coupon->product_categories ),
+			'exclude_product_category_ids' => array_map( 'absint', (array) $coupon->exclude_product_categories ),
+			'exclude_sale_items'           => $coupon->exclude_sale_items(),
+			'minimum_amount'               => wc_format_decimal( $coupon->minimum_amount, 2 ),
+			'maximum_amount'               => wc_format_decimal( $coupon->maximum_amount, 2 ),
+			'customer_emails'              => $coupon->customer_email,
```

Comment:
A few of these fields get renamed in 2.7 (the underlying meta field is the same name, but the name exposed in get/set functions are updated in a few places to make them clearer). https://github.com/woothemes/woocommerce/wiki/2.7-Coupon-Data - some of the other CRUD classes may be doing this as well. I know this is pre-CRUD merge, but it would be annoying to have to version the REST API in the next version to make them match.